pyOCD/target/cortex_m.py

```diff
@@ -488,7 +488,10 @@ def writeMemory(self, addr, value, transfer_size = 32):
         write a memory location.
         By default the transfer size is a word
         """
-        self.transport.writeMem(addr, value, transfer_size)
+        try:
+            self.transport.writeMem(addr, value, transfer_size)
+        except:
+            self.handleMemoryError()
```

Comment:
When this exception occurs, does only the debug logic get reset and not the processor?  Maybe writeMemory could recover from the error by performing the write again.  That way the calling code could continue what is it doing, confident that the write succeed.